tests/Providers/FindologicServiceProviderTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Findologic\Tests\Providers;
+
+use Findologic\Services\SearchService;
+use Plenty\Plugin\ConfigRepository;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
+use Findologic\Providers\FindologicServiceProvider;
+use Findologic\Constants\Plugin;
+use Plenty\Plugin\Events\Dispatcher;
+use Plenty\Plugin\Http\Request;
+
+/**
+ * Class SearchServiceTest
+ * @package Findologic\Tests
+ */
+class FindologicServiceProviderTest extends TestCase
+{
+    /**
+     * @var SearchService|MockObject
+     */
+    protected $searchService;
+
+    /**
+     * @var Request|MockObject
+     */
+    protected $request;
+
+    /**
+     * @var Dispatcher|MockObject
+     */
+    protected $eventDispatcher;
+
+    /**
+     * @var ConfigRepository|MockObject
+     */
+    protected $configRepository;
+
+    public function setUp()
+    {
+        $this->searchService = $this->getMockBuilder(SearchService::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->request = $this->getMockBuilder(Request::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->eventDispatcher = $this->getMockBuilder(Dispatcher::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->configRepository = $this->getMockBuilder(ConfigRepository::class)->disableOriginalConstructor()->setMethods([])->getMock();
+    }
+
+    public function testBootShopKeyNotSet()
+    {
+        $this->configRepository->expects($this->once())->method('get')->willReturn('');
+
+        $this->eventDispatcher->expects($this->never())->method('listen');
+
+        $this->runBoot();
+    }
+
+    public function testBootShopKeySetAndAliveTestFails()
+    {
+        $this->configRepository->expects($this->once())->method('get')->willReturn('testShopKey');
```

Comment:
Explicitly mock the alive test to return something invalid and add a check that it was called once.

Reply:
I have changed to that it mocks throwing exception, because in _FindologicServiceProvider.php_, it is catching the exception and returning, not checking the return value of _aliveTest()_